ServerCore/Helpers/MailHelper.cs

```diff
@@ -0,0 +1,117 @@
+using Microsoft.Extensions.Configuration;
+
+/// <summary>
+/// Helper class for sending mail with Mailjet.
+/// </summary>
+public class MailHelper
+{
+    private string PublicSecret;
+    private string PrivateSecret;
+
+    public bool Enabled { get; private set; }
+    private bool IsDev;
+
+    private const string FromAddress = "msph@puzzlehunt.org";
+    private const string ToAddressForBCC = "msph-bcc@puzzlehunt.org";
+    
+    public static MailHelper Singleton { get; private set; }
+
+    public static void initialize(IConfiguration configuration, bool isDev)
+    {
+        Singleton = new MailHelper(configuration, isDev);
+    }
+
+    private MailHelper(IConfiguration configuration, bool isDev)
+	{
+        try
+        {
+            PublicSecret = configuration["Mailjet-Public"];
+            PrivateSecret = configuration["Mailjet-Private"];
+            Enabled = true;
+            IsDev = isDev;
+        }
+        catch
+        {
+            Enabled = false;
+        }
+    }
+
+    public void Send(string recipient, string subject, string body)
+    {
+        Send(new List<string> { recipient }, false, subject, body);
+    }
+
+    public void Send(IEnumerable<string> recipients, bool bcc, string subject, string body)
+    {
+        // See https://www.mailjet.com/docs/code/c/c-sharp for sample code from Mailjet.
+        if (!Enabled)
+        {
+            if (!IsDev)
+            {
+                throw new InvalidOperationException("Mail was not configured.");
+            }
+            WriteMailToConsole(recipients, subject, body);
+            return;
+        }
+        MailMessage msg = new MailMessage();
+
+        msg.From = new MailAddress(FromAddress);
+
+        if (bcc)
+        {
+            msg.To.Add(new MailAddress(ToAddressForBCC));
+            foreach (string recipient in recipients) {
+                try
+                {
+                    msg.Bcc.Add(new MailAddress(recipient));
+                }
+                catch (FormatException)
+                {
+                    ;  // ignore
+                }
+            }
+        }
+        else
+        {
+            foreach (string recipient in recipients) {
+                try
+                {
+                    msg.To.Add(new MailAddress(recipient));
+                }
+                catch (FormatException)
+                {
+                    ;  // ignore
+                }
+            }
+        }
+
+        msg.Subject = subject;
+        msg.Body = body;
+
+        SmtpClient client = new SmtpClient("in.mailjet.com", 587);
+        // Unlike port 25 which is for general SMTP, port 587 is for
+        // MSA (Message Submission Agent) use and is authenticated.
+        client.DeliveryMethod = SmtpDeliveryMethod.Network;
+        client.EnableSsl = true;
+        client.UseDefaultCredentials = false;
+        client.Credentials = new NetworkCredential(PublicSecret, PrivateSecret);
+        client.Send(msg);
```

Comment:
It might be good to use SendMailAsync here and make MailHelper.Send an async method by changing the signature to ```async Task SendAsync(IEnumerable<string> recipients, bool bcc, string subject, string body)```

Reply:
I haven't used the new async stuff in C# before. I understand the benefit of calling `SendMailAsync` instead. Since this function returns void, why would we need to change it to return a Task instead? Wouldn't the async call just work? What would we do with the Task?

Reply:
Oh. I get it. If I do that then I don't need to wrap the call in `await (Task.Run())`?

Reply:
It works fine if I switch to `SendMailAsync` and remove the `Task.Run`. I'll push it with that change. Let me know if it's wrong.